batch2/batch/worker.py

```diff
@@ -160,13 +168,17 @@ def step(self, name, **kwargs):
         status = {
             'state': cstate['Status'],
             'started_at': cstate['StartedAt'],
-            'finished_at': cstate['FinishedAt']
+            'finished_at': cstate['FinishedAt'],
         }
         cerror = cstate['Error']
         if cerror:
             status['error'] = cerror
         else:
             status['exit_code'] = cstate['ExitCode']
+
+        if cstate['OOMKilled']:
+            status['out_of_memory'] = True
```

Comment:
Nice.  We might set this unconditionally, with False if it didn't run out of memory.  What's the docker container state in this case?  The (Hail) container state?  Are they error?

Reply:
The docker container was exiting with exit code 137. No error. I don't think the logs said it was out of memory error, so I think you have to look at the flag to know docker killed it for out of memory. The Hail container state will be failed.

Reply:
Let me know if you think this field shouldn't be optional.

Reply:
I think it shouldn't be optional.

I'm debating about whether we should return this as failed, exit_code=137 or error, error=Out Of Memory Exception.  out_of_memory is good, it preserves the knowledge of out of memory, but, after years of trying to read the Spark tea leaves, it wouldn't be clearer to users as a proper error message (with exit_code=137).

Let's leave it as you have it now and see if people get confused.  We'll definitely want to document this at some point.  We don't want to leave people to read the tea leaves.